docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
+
+ContainerPilot's core functionality is centered around the concepts of jobs, the events that trigger those jobs or that jobs emit, and watches for the external service discovery catalog (Consul).
```

Comment:
I guess also you're doing that within the next section as well, intermingled with jobs themselves. Which also makes sense.